src/main/java/com/denizenscript/ddiscordbot/commands/DiscordModalCommand.java

```diff
@@ -0,0 +1,133 @@
+import net.dv8tion.jda.api.requests.restaction.interactions.ModalCallbackAction;
+import org.bukkit.Bukkit;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class DiscordModalCommand extends AbstractDiscordCommand implements Holdable {
+
+    public DiscordModalCommand() {
+        setName("discordmodal");
+        setSyntax("discordmodal [interaction:<interaction>] [name:<name>] [rows:<rows>] (title:<title>)");
+        setRequiredArguments(3, 4);
+        setPrefixesHandled("interaction", "rows", "title", "name");
+        isProcedural = false;
+    }
+
+    // <--[command]
+    // @Name discordmodal
+    // @Syntax discordmodal [interaction:<interaction>] [name:<name>] [rows:<rows>] (title:<title>)
+    // @Required 3
+    // @Maximum 4
+    // @Short Manages Discord modals
+    // @Plugin dDiscordBot
+    // @Guide https://guide.denizenscript.com/guides/expanding/ddiscordbot.html
+    // @Group external
+    //
+    // With this command you can respond to an interaction using a modal.
+    //
+    // You should usually defer an interaction before using a modal.
+    //
+    // The command should usually be ~waited for. See <@link language ~waitable>.
+    //
+    // @Tags
+    // <entry[saveName].command> returns the DiscordCommandTag of a slash command upon creation, when the command is ~waited for.
+    //
+    // @Usage
+    // Use to create a modal from text replies.
+    // - definemap rows:
+    //      1:
+    //          1: <discord_text_input.with[name].as[textinput].with[label].as[Type here!].with[style].as[short]>
+    // - ~discordmodal interaction:<context.interaction> name:example_modal title:Modal! rows:<[rows]>
+    //
+    // -->
+
+    @Override
+    public void execute(ScriptEntry scriptEntry) {
+        DiscordInteractionTag interaction = scriptEntry.requiredArgForPrefix("interaction", DiscordInteractionTag.class);
+        ElementTag name = scriptEntry.requiredArgForPrefixAsElement("name");
+        ElementTag title = scriptEntry.argForPrefixAsElement("title", "");
+        ObjectTag rows = scriptEntry.requiredArgForPrefix("rows", ObjectTag.class);
+        if (scriptEntry.dbCallShouldDebug()) {
+            Debug.report(scriptEntry, getName(), interaction, name, rows);
+        }
+        Runnable runner = () -> {
+            try {
+                if (interaction.interaction == null) {
+                    handleError(scriptEntry, "Invalid interaction! Has it expired?");
+                    return;
+                }
+                else if (rows == null) {
+                    handleError(scriptEntry, "Must have action rows!");
+                    return;
+                }
+                else if (name == null) {
+                    handleError(scriptEntry, "Must have a name!");
+                    return;
+                }
+                List<ActionRow> actionRows = DiscordModalCommand.createRows(scriptEntry, rows);
```

Comment:
`DiscordModalCommand.` should not be here - you're calling a method in the same class